packages/flutter/lib/src/foundation/constants.dart

```diff
@@ -30,6 +44,20 @@ const bool kProfileMode = bool.fromEnvironment('dart.vm.profile', defaultValue:
 /// Since this is a const value, it can be used to indicate to the compiler that
 /// a particular block of code will not be executed in debug mode, and hence
 /// can be removed.
+///
+/// An alternative strategy is to use asserts, as in:
+///
+/// ```dart
```

Comment:
Turn this into a {@tool snippet} so it gets analyzed?

Reply:
done

Reply:
had to revert this, the snippet analysis logic doesn't like bare asserts.